File: java/src/Leetcode1768.java
class Leetcode1768{
    public String mergeAlternately(String word1, String word2) {
        StringBuilder merged = new StringBuilder();
        int i = 0, j = 0;
        while(i < word1.length() && j < word2.length()){
            if(i == j){
                merged.append(word1.charAt(i++));                                
            }else{
                merged.append(word2.charAt(j++));
            }
        }

        while(i < word1.length()){
            merged.append(word1.charAt(i++));
        }
        while(j < word2.length()){
            merged.append(word2.charAt(j++));
        }

        return merged.toString();
    }
}

File: python/Leetcode1768.py
class Leetcode1768:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merged = ""
        a, b = 0, 0
        while a < len(word1) and b < len(word2):
            if a == b:
                merged += word1[a]
                a += 1
            else:
                merged += word2[b]
                b += 1
        if a < len(word1):
            merged += word1[a:]
        if b < len(word2):
            merged += word2[b:]            
        return merged